lib/ui/src/components/sidebar/Search.tsx

```diff
@@ -303,10 +324,14 @@ export const Search = React.memo<{
           getRootProps,
           highlightedIndex,
         }) => {
-          const input = inputValue ? inputValue.trim() : '';
-          let results: DownshiftItem[] = input ? getResults(input) : [];
+          const isBrowsing = !isOpen && document.activeElement !== inputRef.current;
+          const filter = inputValue ? inputValue.trim() : '';
+          let results: DownshiftItem[] = filter ? getResults(filter) : [];
+
+          syncUrlToFilter(isBrowsing ? null : filter);
+          api.setQueryParams({ filter: isBrowsing ? null : filter });
```

Comment:
I think we don't even need `syncUrlToFilter` and can just rely on `api.setQueryParams` to do the `replaceState` for us. The only thing is the value cannot be `null` because `setQueryParams` will ignore those. Probably you can just set it to `undefined`, but I'm not exactly sure about its behavior.

Reply:
Are saying removing `syncUrlToFilter` will satisfy the feature then? I'm sure I put it in for a reason but it seems so long ago now 😅 

I can take it out and try to get the CI playing ball again 👍